src/Runner.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Pantheon\Terminus;
+
+use Consolidation\AnnotatedCommand\AnnotatedCommandFactory;
+use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+
+class Runner
+{
+    /**
+     * @var Terminus
+     */
+    private $application;
+    /**
+     * @var string
+     */
+    private $commands_directory = __DIR__ . '/Commands';
+    /**
+     * @var string
+     */
+    private $top_namespace = 'Pantheon\Terminus\Commands';
+
+    /**
+     * Runner constructor
+     *
+     * @param array $options Options to configure the runner
+     */
+    public function __construct(array $options = [])
+    {
+        $this->application = new Terminus();
+        $this->configureApplication(new Config($options));
+    }
+
+    /**
+     * Runs the instantiated Terminus application
+     */
+    public function run()
+    {
+        $this->application->run();
+    }
+
+    /**
+     * Adds command files to the application
+     *
+     * @param Config $config A Terminus configuration object
+     * @return void
+     */
+    private function configureApplication(Config $config)
+    {
+        $command_factory = new AnnotatedCommandFactory();
+        $discovery = new CommandFileDiscovery();
+        $discovery->setSearchPattern('*Command.php');
+        $subdirectories = array_merge(
+            $this->locateSubdirectories($this->commands_directory, $this->top_namespace),
+            $this->locateSubdirectories($config->get('plugins_dir'))
+        );
+        foreach ($subdirectories as $subdirectory) {
+            $command_files = $discovery->discover($subdirectory['path'], $subdirectory['namespace']);
+            foreach ($command_files as $command_class) {
+                $reflection_class = new \ReflectionClass($command_class);
+                if (!$reflection_class->isAbstract()) {
+                    $command_list = $command_factory->createCommandsFromClass(new $command_class($config));
+                    foreach ($command_list as $command) {
+                        $this->application->add($command);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Recursively discovers all directories underneath a given directory
+     *
+     * @param $directory Directory to start search from
+     * @param $namespace Namespace for the given directory
+     */
+    private function locateSubdirectories($directory, $namespace = '\\')
```

Comment:
Thisis what CommandFileDiscovery will do for you. I think the default search location is `src`, but you can set that to something else if you want it to search in all subdirectories.